http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala

```diff
@@ -520,99 +531,84 @@ class RuntimeV2ServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
   )(implicit as: Ask[F, AppContext]): F[Vector[ListRuntimeResponse2]] =
     for {
       ctx <- as.ask
+
+      // Parameters: parse search filters from request
       (labelMap, includeDeleted, _) <- F.fromEither(processListParameters(params))
       excludeStatuses = if (includeDeleted) List.empty else List(RuntimeStatus.Deleted)
       creatorOnly <- F.fromEither(processCreatorOnlyParameter(userInfo.userEmail, params, ctx.traceId))
+
+      // Authorize: user has an active account and has accepted terms of service
+      _ <- authProvider.checkUserEnabled(userInfo)
+
+      // Authorize: get resource IDs the user can see
+      // HACK: leonardo is modeling access control here, handling inheritance
+      // of workspace and project-level permissions. Sam and WSM already do this,
+      // and should be considered the point of truth.
+
+      // HACK: leonardo short-circuits access control to grant access to runtime creators.
+      // This supports the use case where `terra-ui` requests status of runtimes that have
+      // not yet been provisioned in Sam.
+      creatorRuntimeIdsBackdoor <-
+        if (creatorOnly.isDefined)
+          RuntimeServiceDbQueries
+            .listRuntimeIdsForCreator(creatorOnly.get)
+            .map(_.map(_.samResource).toSet)
+            .transaction
+            .flatMap(F.pure(_))
+        else
+          F.pure(Set.empty)
+
+      // v1 runtimes (sam resource type `notebook-cluster`) are readable only
+      // by their creators (`Creator` is the SamResource.Runtime `ownerRoleName`),
+      // if the creator also has read access to the corresponding SamResource.Project
+      creatorV1RuntimeIds: Set[RuntimeSamResourceId] <- authProvider
+        .listResourceIds[RuntimeSamResourceId](hasOwnerRole = true, userInfo)
+      readerProjectIds: Set[ProjectSamResourceId] <- authProvider
+        .listResourceIds[ProjectSamResourceId](hasOwnerRole = false, userInfo)
```

Comment:
Don't need to fix for this PR, but I think this is essentially making 2 identical `/api/resources/v2/googleProject` Sam calls, and filtering the response differently based on `hasOwnerRole`. Could potentially optimize so that it just makes 1 Sam call per resource type.

Reply:
That's true; same for workspace. I don't think it's any worse than before so i will keep it in mind as a future optimization!